controllers/virtualmachinegroup/virtualmachinegroup_controller.go

```diff
@@ -912,3 +932,58 @@ func aggregateOrNoRequeue(errs []error) error {
 
 	return agg
 }
+
+// ensurePlacementStatusUpToDate checks if all members defined in spec have the
+// PlacementReady condition true in status.members. If any member doesn't have
+// the PlacementReady condition true, it uses the api reader to get a fresh copy
+// of the group object to avoid stale placement status.
+func (r *Reconciler) ensurePlacementStatusUpToDate(
+	ctx context.Context,
+	vmGroup *vmopv1.VirtualMachineGroup) (*vmopv1.VirtualMachineGroup, error) {
+
+	// Build a map of all VM-kind members from the group's spec.
+	specVMMembers := make(map[string]struct{})
+	for _, bootOrder := range vmGroup.Spec.BootOrder {
+		for _, member := range bootOrder.Members {
+			if member.Kind == vmKind {
+				specVMMembers[member.Name] = struct{}{}
+			}
+		}
+	}
+
+	// Build a map of all VM-kind members from the group's status.
```

Comment:
Shouldn't this consider all member kinds, not just VMs?

Reply:
> Should we set that condition on the group itself for when all of its direct members have a placement ready condition?

That's included as part of the group's overall Ready condition. It will not be true unless all direct members have PlacementReady true.

Reply:
Sure, but a separate condition, specifically about all members having placement, but _also_ allow for each child-group to be considered, right?